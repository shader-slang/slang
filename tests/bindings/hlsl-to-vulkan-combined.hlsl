//TEST:REFLECTION:-target glsl -profile ps_4_0 -entry main -fvk-t-shift 5 all -fvk-t-shift 7 2  -fvk-s-shift -3 0 -fvk-b-shift 1 2

Sampler2D<float> t0 : register(t2);
Sampler2D<float4> t1 : register(t7, space2);

float4 main() : SV_TARGET
{
	return float4(1, 1, 1, 0);
}