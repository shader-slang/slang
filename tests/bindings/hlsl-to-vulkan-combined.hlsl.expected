result code = 0
standard error = {
}
standard output = {
{
    "parameters": [
        {
            "name": "t0",
            "binding": {"kind": "descriptorTableSlot", "index": 0},
            "type": {
                "kind": "resource",
                "baseShape": "texture2D"
            }
        },
        {
            "name": "t1",
            "binding": {"kind": "descriptorTableSlot", "index": 1},
            "type": {
                "kind": "resource",
                "baseShape": "texture2D"
            }
        }
    ],
    "entryPoints": [
        {
            "name": "main",
            "stage:": "fragment",
            "result:": {
                "stage": "fragment",
                "binding": {"kind": "varyingOutput", "index": 0},
                "semanticName": "SV_TARGET",
                "type": {
                    "kind": "vector",
                    "elementCount": 4,
                    "elementType": {
                        "kind": "scalar",
                        "scalarType": "float32"
                    }
                }
            },
            "bindings": [
                {
                    "name": "t0",
                    "binding": {"kind": "descriptorTableSlot", "index": 0, "used": 0}
                },
                {
                    "name": "t1",
                    "binding": {"kind": "descriptorTableSlot", "index": 1, "used": 0}
                }
            ]
        }
    ]
}
}
